Source/CharismaModule/Private/CharismaAPI.cpp

```diff
@@ -0,0 +1,196 @@
+#include "CharismaAPI.h"
+#include "Playthrough.h"
+
+#include "Json.h"
+#include "JsonUtilities.h"
+
+#include <nlohmann/json.hpp>
+#include <sstream>
+
+const FString UCharismaAPI::BaseURL = TEXT("https://play.charisma.ai");
+const FString UCharismaAPI::SocketURL = UCharismaAPI::BaseURL.Replace(TEXT("http"), TEXT("ws"));
+
+FString UCharismaAPI::ToQueryString(const TMap<FString, FString>& QueryParams)
+{
+	if (!QueryParams.Num())
+	{
+		return TEXT("");
+	}
+
+	FString Result;
+	for (const TPair<FString, FString>& Pair : QueryParams)
+	{
+		Result.Append(Pair.Key);
+		Result.Append(TEXT(""));
+		Result.Append(Pair.Value);
+	}
+
+	return Result;
+}
+
+void UCharismaAPI::CreateCharismaPlaythroughObject(UPlaythrough*& Playthrough)
+{
+	Playthrough = NewObject<UPlaythrough>();
+}
```

Comment:
This could make more sense to be on the `UPlaythrough` class as `UPlaythrough::CreatePlaythroughObject`, as it's more to do with the Playthrough than the API 🙂